src/iquery.cpp

```diff
@@ -0,0 +1,173 @@
+#include <cstdlib>
+#include <iostream>
+#include <iterator>
+
+auto print_usage_info() -> void;
+auto print_column_mappings(const char* _json) -> void;
+
+int main(int _argc, char* _argv[]) // NOLINT(modernize-use-trailing-return-type)
+{
+    set_ips_display_name("iquery");
+
+    namespace po = boost::program_options;
+
+    po::options_description desc{""};
+
+    // clang-format off
+    desc.add_options()
+        ("columns,c", po::bool_switch(), "")
+        ("query_string", po::value<std::string>()->default_value("-"), "")
+        ("sql-only", po::bool_switch(), "")
+        ("zone,z", po::value<std::string>(), "")
+        ("help,h", "");
+    // clang-format on
+
+    po::positional_options_description pod;
+    pod.add("query_string", 1);
+
+    load_client_api_plugins();
+
+    try {
+        po::variables_map vm;
+        po::store(po::command_line_parser(_argc, _argv).options(desc).positional(pod).run(), vm);
+        po::notify(vm);
+
+        if (vm.count("help")) {
+            print_usage_info();
+            return 0;
+        }
+
+        Genquery2Input input{};
+        std::string query_string;
+
+        if (vm["columns"].as<bool>()) {
+            input.column_mappings = 1;
+        }
+        else {
+            query_string = vm["query_string"].as<std::string>();
+
+            // Read from stdin.
+            if ("-" == query_string) {
+                query_string.clear();
+                std::getline(std::cin, query_string);
+            }
+
+            if (query_string.empty()) {
+                fmt::print(stderr, "error: Missing QUERY_STRING\n");
+                return 1;
+            }
+
+            input.query_string = query_string.data();
+        }
+
+        std::string zone;
+        if (vm.count("zone")) {
+            zone = vm["zone"].as<std::string>();
+            input.zone = zone.data();
+        }
+
+        if (vm["sql-only"].as<bool>()) {
+            input.sql_only = 1;
+        }
+
+        irods::experimental::client_connection conn;
+        char* output{};
+        irods::at_scope_exit free_sql{[&output] {
+            if (output) {
+                std::free(output);
+            }
+        }};
+
+        const auto ec = rc_genquery2(static_cast<RcComm*>(conn), &input, &output);
+        if (ec < 0) {
+            fmt::print(stderr, "error: {}\n", ec);
+            return 1;
+        }
+
+        if (1 == input.column_mappings) {
+            print_column_mappings(output);
+            return 0;
+        }
+
+        fmt::print("{}\n", output);
+
+        return 0;
+    }
+    catch (const irods::exception& e) {
+        fmt::print(stderr, "error: {}\n", e.client_display_what());
+    }
+    catch (const std::exception& e) {
+        fmt::print(stderr, "error: {}\n", e.what());
+    }
+
+    return 1;
+} // main
+
+auto print_usage_info() -> void
+{
+    fmt::print(R"_(iquery - Query the iRODS Catalog
+
+Usage: iquery [OPTION]... QUERY_STRING
+
+Queries the iRODS Catalog using GenQuery2.
```

Comment:
Is this iCommand considered experimental given that the API it calls is experimental? Should we mention that in the help text?

Reply:
The help text originally included `(experimental)` beside the icommand name.

Reverting that change makes sense given the actual API could change. Although one could argue the interface of `iquery` is independent of the underlying API.

Reply:
Fair enough. I think it's fine as-is, then

Reply:
The safest thing to do is to revert the removal of the `(experimental)` tag.

Will think about that a little.

Reply:
Decided to restore the experimental tag to indicate to users that things can change in the future.